### ISSUE TITLE
Banner Graphic

### ISSUE THREAD

**Issue by NicoledeGreef** [bcgov/smk-legacy-ui#49]
Allow for optional inclusion of a banner graphic...

**Comment by ll911** [comment]
standalone webapp include map should all comply with GCPE's look/feel
https://github.com/bcgov/Gov-2.0-Bootstrap-Skeleton 
and here is the sample
https://angular-scaffold-demo.pathfinder.gov.bc.ca/

**Comment by qqnluaq** [comment]
added option to admin tool to configure an image for the surround.

**Comment by cpspicer** [comment]
Can't seem to get a image to display.  No 'Keep changes' option in the author interface.  I hit save and a message comes up looking for attachments.  Filename is preserved in the Style & Theme tab if I re-edit the application, but the image does not replace the standard blue header with gov logo.

**Comment by cpspicer** [comment]
Related to file upload issues on OpenShift?

**Comment by dhlevi** [comment]
Currently the banner graphic is an icon replacement for the BC Gov logo on the left hand side. This is functioning now as an attachment in the map config. 

We've extended this functionality a bit in the latest release, but to clarify the desired outcome, should this be a full header bar replacement image (like DMF?)

**Comment by cpspicer** [comment]
When I upload an image, the site name moves to the right in the banner, but image isn't displayed.  Tried .JPEG, .PNG, .GIF.  Admin notes attachments uploaded successfully.  Sample images appended below:

![my outlook profile pic](https://user-images.githubusercontent.com/26190669/40754231-0d42402c-642d-11e8-8512-36f950087c05.GIF)
![my outlook profile pic](https://user-images.githubusercontent.com/26190669/40754235-16f6c4f8-642d-11e8-8085-c949632343a0.JPG)

**Comment by cpspicer** [comment]
Need a screen sharing session - still can't get it to work.

**Comment by dhlevi** [comment]
Just a thought: Have you done a hard-refresh of your browser, to ensure the old cache is no longer present?

Also, please see my note in issue #12 regarding the banner.

**Comment by cpspicer** [comment]
Did a hard clear.  Works 50% of the time, after 6 attempts.  Half the time I'd get an error uploading attachment.  Found the successful attempts included the following steps:  1) Click on themes tab 2) Click on  Style and Theme 3) Click on File and choose one 4) click back.  5) click next.  6) click save.

Doing a straight save right after choosing the file always resulted in a failed attempt and generated an 'upload' error.

**Comment by dhlevi** [comment]
Hmm... I've just tried
- Click Edit
- Click themes Tab
- Click file, choose one
- Click save

Several times on the test examples I'm working on, and all of them succeeded, no errors thrown. So there's something happening different on your end than what I can see over here.

Questions: 

Which browser are you using: Firefox? Chrome? Edge? IE11?
Are you using: https://smk-demo.pathfinder.gov.bc.ca/smk-ui/ or http://smk.pathfinder.gov.bc.ca/smk-ui/

**Comment by cpspicer** [comment]
Using Version 67.0.3396.62 (Official Build) (64-bit)  Chrome

http://smk.pathfinder.gov.bc.ca/smk-ui/#

**Comment by dhlevi** [comment]
OK. I tried the same process as usual (described above), using chrome (version 66.0.3359.181 mind you, apparently I need an update) and the http url as you are, and it worked for me every time. 

I was using several of my own test images, as well as the two you linked above. Aside from a script caching problem, I'm not 100% sure exactly how I can't make it fail, and you can't get it to work... 

Just to verify, you did a full hard reset on the cache? In chrome, it's best to open the "Inspect" tab by right-clicking anywhere on the page or hitting ctrl+shift+I, then right-clicking again on the refresh button and selecting "Empty Cache and Hard Reload"

**Comment by cpspicer** [comment]
Yes, verified.  Also moved the image being uploaded directly to my C:\ drive and off the network share.  I'll grab another PC that's connected outside the gov network and try again. 

![image](https://user-images.githubusercontent.com/26190669/41008510-30b97d60-68e0-11e8-95b3-51ac531c368f.png)

**Comment by cpspicer** [comment]
No luck using a non-gov't laptop with external network connection; also no luck trying IE.  Feels like something with the Openshift environment...
